src/circular_buffer.jl

```diff
@@ -1,22 +1,31 @@
 """
-    CircularBuffer{T}(n)
+    CircularBuffer{T}(v,n)
 
 The CircularBuffer type implements a circular buffer of fixed capacity
 where new items are pushed to the back of the list, overwriting values
 in a circular fashion.
 
-Allocate a buffer of elements of type `T` with maximum capacity `n`.
+Allocate a buffer of elements of type `T`  containing items `v` with maximum capacity `n`.
 """
 mutable struct CircularBuffer{T} <: AbstractVector{T}
     capacity::Int
     first::Int
     length::Int
     buffer::Vector{T}
 
-    CircularBuffer{T}(capacity::Int) where {T} = new{T}(capacity, 1, 0, Vector{T}(undef, capacity))
+    function CircularBuffer{T}(vec,capacity::Int) where {T}
+        capacity >= length(vec) || throw(BoundsError()) # prevent overflow
```

Comment:
We don't need to guard gainst this. `copyto!` guards against it for us.
Which si good because that means we can argange to avoid calling `length` on an iterator which might not have `length` defined`